src/lib.rs

```diff
@@ -13,21 +16,93 @@ fn get_user_input(prompt: &str) -> String {
 }
 
 fn take_guess() -> char {
-    return get_user_input("Provide the character you want to guess: ")
+    // take a single character as guess input
+    return take_user_input("Provide the character you want to guess: ")
         .chars()
         .next()
         .expect("More than one character was guessed.");
 }
 
+const HANGMAN_STAGES: [&str; 7] = [
+    "
+       +---+
+       |   |
+           |
+           |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+           |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+       |   |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|   |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|\\  |
+           |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|\\  |
+      /    |
+           |
+     =========",
+    "
+       +---+
+       |   |
+       O   |
+      /|\\  |
+      / \\  |
+           |
+     =========",
+];
+
+fn print_hangman_stage(incorrect_guesses: usize) {
+    if incorrect_guesses < HANGMAN_STAGES.len() {
+        println!("{}", HANGMAN_STAGES[incorrect_guesses]);
+    } else {
+        panic!("Invalid number of incorrect guesses.");
+    }
+}
+
+// game core
+
 struct PlayerState {
-    wrong_guesses: u32,
+    wrong_guesses: usize,
 }
 
 fn run_game_loop(config: &mut Config, guess: char, player_state: &mut PlayerState) -> GameState {
     if !config.secret_word.contains(guess) {
         player_state.wrong_guesses += 1;
 
-        if player_state.wrong_guesses > config.lives {
+        print_hangman_stage(player_state.wrong_guesses + (6 - config.lives));
+
```

Comment:
[nit]: We could think about simplifying this if we set the number of lives to 6:
```suggestion
        print_hangman_stage(player_state.wrong_guesses);

```